bolt-sidecar/src/builder/mod.rs

```diff
@@ -62,97 +70,109 @@ pub enum BuilderError {
 pub struct LocalBuilder {
     /// BLS credentials for the local builder. We use this to sign the
     /// payload bid submissions built by the sidecar.
-    secret_key: BlsSecretKey,
+    secret_key: SecretKey,
+    /// Chain configuration
+    /// (necessary for signing messages with the correct domain)
+    chain: Chain,
     /// Async fallback payload builder to generate valid payloads with
     /// the engine API's `engine_newPayloadV3` response error.
     fallback_builder: FallbackPayloadBuilder,
-    /// Cached payloads by block hash. This is used to respond to
-    /// the builder API `getPayload` requests with the full block.
-    cached_payloads: HashMap<B256, ExecutionPayload>,
+    /// The last payload and bid that was built by the local builder.
+    payload_and_bid: Option<PayloadAndBid>,
 }
 
 impl LocalBuilder {
     /// Create a new local builder with the given secret key.
-    pub fn new(
-        secret_key: BlsSecretKey,
-        execution_rpc_url: &str,
-        engine_rpc_url: &str,
-        engine_jwt_secret: &str,
-        fee_recipient: Address,
-    ) -> Self {
+    pub fn new(config: &Config) -> Self {
         Self {
-            secret_key,
-            cached_payloads: Default::default(),
-            fallback_builder: FallbackPayloadBuilder::new(
-                engine_jwt_secret,
-                fee_recipient,
-                execution_rpc_url,
-                engine_rpc_url,
-            ),
+            payload_and_bid: None,
+            fallback_builder: FallbackPayloadBuilder::new(config),
+            secret_key: config.builder_private_key.clone(),
+            chain: config.chain.clone(),
         }
     }
 
     /// Build a new payload with the given transactions. This method will
-    /// return a signed builder bid that can be submitted to the Builder API.
-    pub async fn build_new_payload(
+    /// cache the payload in the local builder instance, and make it available
+    ///
+    pub async fn build_new_local_payload(
         &mut self,
         transactions: Vec<TransactionSigned>,
-    ) -> Result<SignedBuilderBid, BuilderError> {
+    ) -> Result<(), BuilderError> {
         // 1. build a fallback payload with the given transactions, on top of
         // the current head of the chain
         let sealed_block = self
             .fallback_builder
-            .build_fallback_payload(transactions)
+            .build_fallback_payload(&transactions)
             .await?;
 
         // NOTE: we use a big value for the bid to ensure it gets chosen by mev-boost.
         // the client has no way to actually verify this, and we don't need to trust
         // an external relay as this block is self-built, so the fake bid value is fine.
         let value = U256::from(1_000_000_000_000_000_000u128);
```

Comment:
Note: we don't strictly need this. The validator & beacon nodes have options to ALWAYS prefer PBS blocks (but let's leave it for now to be sure)

Reply:
Good point, added this to the note comment as I think it's useful to remember